src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -86,12 +86,19 @@ protected override void InitializeWorker(CompilationStartAnalysisContext context
             var charType = context.Compilation.GetSpecialType(SpecialType.System_Char);
             var boolType = context.Compilation.GetSpecialType(SpecialType.System_Boolean);
             var guidType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemGuid);
+            var nullableTType = context.Compilation.GetSpecialType(SpecialType.System_Nullable_T);
+            INamedTypeSymbol? GetNullableType(INamedTypeSymbol? typeSymbol) => typeSymbol is not null ? nullableTType?.Construct(typeSymbol) : null;
 
             var builder = ImmutableHashSet.CreateBuilder<INamedTypeSymbol>();
+
             builder.AddIfNotNull(charType);
+            builder.AddIfNotNull(GetNullableType(charType));
             builder.AddIfNotNull(boolType);
+            builder.AddIfNotNull(GetNullableType(boolType));
             builder.AddIfNotNull(stringType);
             builder.AddIfNotNull(guidType);
+            builder.AddIfNotNull(GetNullableType(guidType));
```

Comment:
Would it be better to do the fix in `IsValidToStringCall`? (i.e, instead of constructing `char?`, `bool?`, and `Guid?`, just try to find `Nullable<SomeType>` and see if `SomeType` is in the hashset.

Reply:
Good idea. I went ahead and did that.